spring-tx/src/main/java/org/springframework/transaction/reactive/TransactionalOperatorImpl.java

```diff
@@ -71,6 +71,21 @@ public ReactiveTransactionManager getTransactionManager() {
 		return this.transactionManager;
 	}
 
+	@Override
+	public <T> Mono<T> transactional(Mono<T> mono) {
+		return TransactionContextManager.currentContext().flatMap(context -> {
+			Mono<ReactiveTransaction> status = this.transactionManager.getReactiveTransaction(this.transactionDefinition);
+			// This is an around advice: Invoke the next interceptor in the chain.
+			// This will normally result in a target object being invoked.
+			// Need re-wrapping of ReactiveTransaction until we get hold of the exception
+			// through usingWhen.
+			return status.flatMap(it -> Mono.usingWhen(Mono.just(it), ignore -> mono,
+					this.transactionManager::commit, s -> Mono.empty())
```

Comment:
note that in case of cancellation, since no `asyncCancel` is defined here, it defaults to `commit`. Also, it feels weird that the error case is basically ignored

Reply:
Commit on cancel is what we want if just a subset of data is processed. In imperative code this compares to processing a subset of a result list.

The error case is ignored because errors are handled after the `usingWhen` operator. In case a pre-commit action fails, we want to issue a rollback. Committing a transaction consists of several activities of which the actual commit is just one of them. So if a commit fails, we still need to rollback,

Reply:
makes sense, thanks for the explanations. nothing to add 👍